File: frontend/lib/types/recipe.ts
export interface RecipeTypeDto {
  id: number;
  name: string;
}

export interface RecipeCategoryDto {
  id: string;
  name: string;
}

export interface RecipeListItemDto {
  id: string;
  name: string;
  url?: string | null;
  recipeType: RecipeTypeDto;
  categories: RecipeCategoryDto[];
}

export interface RecipeDetailDto extends RecipeListItemDto {
  ingredients?: string | null;
  instructions?: string | null;
}

export interface CreateRecipeInput {
  name: string;
  url?: string;
  ingredients?: string;
  instructions?: string;
}

export const RECIPE_TYPES = {
  external: { id: 1, name: "Externt" },
  own: { id: 2, name: "Eget" },
} as const;

export function isExternalRecipe(recipe: RecipeListItemDto): boolean {
  return recipe.recipeType.id === RECIPE_TYPES.external.id;
}

export function toListItem({
  ingredients: _ingredients,
  instructions: _instructions,
  ...listItem
}: RecipeDetailDto): RecipeListItemDto {
  return listItem;
}

File: backend/RecipeApi/Entities/Recipe.cs
namespace RecipeApi.Entities;

public class Recipe
{
    public Guid Id { get; set; }

    public string Name { get; set; } = string.Empty;

    public string? Url { get; set; }

    public int RecipeTypeId { get; set; }

    public string? Ingredients { get; set; }

    public string? Instructions { get; set; }

    public string? Notes { get; set; }

    public ICollection<Category> Categories { get; set; } = [];
}
